### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by kensteph**
In this pull request i completed the following requirements:

**Create models**
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Remember that author_id column in posts table should be the foreign_key for the users table.
- Use models to insert data using the rails console

**Create custom methods**
- Your models should include some additional methods.

**Users**

- A method that returns the 3 most recent posts for a given user.

**Posts**

- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

**Comments**

- A method that updates the comments counter for a post.

**Likes**

- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Comment < ApplicationRecord
+  belongs_to :post
```
Comment (t-yanick) [inline app/models/comment.rb:2]: - [ ] Great job working working on this milestone. I think you have a repetition of this association on lines 2 and 4. kindly delete one of them.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+  belongs_to :post
+  belongs_to :author, class_name: 'User'
+  after_save :update_post_comments_counter
```
Comment (t-yanick) [inline app/models/comment.rb:4]: - [ ] Great job using the Active Rails Callback - after_save, but I think you're missing the private method just as you did with the post model. Kindly add this private method for update_post_comments_counter

```
priavate  

def update_post__comments_counter
    author.increment!(:post_comments_counter)
  end

```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @kensteph , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great use of Active Rails Callback - after_save
:heavy_check_mark: Descriptive PR and commit messages
:heavy_check_mark: All models added as per project requirements

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 2 above)
Comment (kensteph) [reply app/models/comment.rb:4]: @t-yanick i added it

**Comment by Kamba56** [comment]
Hi @kensteph

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that **does not pass all linters checks.** Kindly fix all linters errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED 🥇 

Hi @kensteph , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All required changes implemented
:heavy_check_mark: Linter checks passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**